napari/plugins/exceptions.py

```diff
@@ -1,23 +1,251 @@
+        if contact:
+            extra = [f'{k: >11}: {v}' for k, v in contact.items()]
+            extra += [f'{"napari": >11}: v{__version__}']
+            msg += "\n".join(extra)
+        msg += '\n'
+        return msg
+
+    def info(
+        self,
+    ) -> Tuple[Type['PluginError'], 'PluginError', Optional[TracebackType]]:
+        """Return info as would be returned from sys.exc_info()."""
+        return (self.__class__, self, self.__traceback__)
 
 
 class PluginImportError(PluginError, ImportError):
     """Raised when a plugin fails to import."""
 
     def __init__(self, plugin_name: str, plugin_module: str) -> None:
-        msg = f'Failed to import plugin: "{plugin_name}"'
+        msg = f"Failed to import plugin: '{plugin_name}'"
         super().__init__(msg, plugin_name, plugin_module)
 
 
 class PluginRegistrationError(PluginError):
     """Raised when a plugin fails to register with pluggy."""
 
     def __init__(self, plugin_name: str, plugin_module: str) -> None:
-        msg = f'Failed to register plugin: "{plugin_name}"'
+        msg = f"Failed to register plugin: '{plugin_name}'"
         super().__init__(msg, plugin_name, plugin_module)
+
+
+def format_exceptions(plugin_name: str, as_html: bool = False) -> str:
+    """Return formatted tracebacks for all exceptions raised by plugin.
+
+    This method takes advantage of ``IPython.core.ultratb.VerboseTB`` to create
+    a nicely formatted traceback.  If ``as_html`` is True, is then converted
+    from ansi to html.  (This approach was used instead of cgitb because the
+    IPython ultratb tracebacks were preferred).
+
+    Parameters
+    ----------
+    plugin_name : str
+        The name of a plugin for which to retrieve tracebacks.
+    as_html : bool
+        Whether to return the exception string as formatted html,
+        defaults to False.
+
+    Returns
+    -------
+    str
+        A formatted string with traceback information for every exception
+        raised by ``plugin_name`` during this session.
+    """
+    from napari import __version__
+
+    if not PLUGIN_ERRORS.get(plugin_name):
+        return ''
+
+    color = 'Linux' if as_html else 'NoColor'
+    vbtb = IPython.core.ultratb.VerboseTB(color_scheme=color)
+
+    _linewidth = 80
+    _pad = (_linewidth - len(plugin_name) - 18) // 2
+    msg = [
+        f"{'=' * _pad} Errors for plugin '{plugin_name}' {'=' * _pad}",
+        '',
+        f'{"napari version": >16}: {__version__}',
+    ]
+    try:
+        err0 = PLUGIN_ERRORS[plugin_name][0]
+        package_meta = fetch_module_metadata(err0.plugin_module)
+        if package_meta:
+            msg.extend(
+                [
+                    f'{"plugin name": >16}: {package_meta["name"]}',
+                    f'{"version": >16}: {package_meta["version"]}',
+                    f'{"module": >16}: {err0.plugin_module}',
+                ]
+            )
+    except Exception:
+        pass
```

Comment:
What exactly are the things that can go wrong here?

Reply:
I never actually saw any exceptions, so it's hard to say.  The reason for this is that the _last_ thing I want to happen is that a real exception get burried due to some unexpected exception raised while trying to get package metadata just to make a nicer error message.

Reply:
But that's exactly what's going to happen, no? All kinds of bugs might be hidden by having a bare except here. Personally, I think we should remove this until we actually observe such an exception. Then we can either fix the cause of that exception or make a specific catch, as appropriate.